app/uk/gov/hmrc/nisp/views/Main.scala

```diff
@@ -0,0 +1,135 @@
+import uk.gov.hmrc.nisp.controllers.routes
+import uk.gov.hmrc.nisp.models.admin.SCAWrapperToggle
+import uk.gov.hmrc.nisp.views.html.oldMain
+import uk.gov.hmrc.nisp.views.html.components.{additionalScripts, additionalStylesheets}
+import uk.gov.hmrc.play.http.HeaderCarrierConverter
+import uk.gov.hmrc.ptafrontend.views.html.PtaHead
+import uk.gov.hmrc.sca.models.BannerConfig
+import uk.gov.hmrc.sca.services.WrapperService
+
+import javax.inject.Inject
+import scala.concurrent.Await
+import scala.concurrent.duration.{Duration, SECONDS}
+import scala.util.{Failure, Success, Try}
+
+@ImplementedBy(classOf[MainImpl])
+trait Main {
+  def apply(
+             pageTitle: String,
+             showUrBanner: Boolean = true,
+             extendedTitle: Boolean = true,
+             optCustomLayout: Option[Html => Html] = None,
+             sidebar: Option[Html] = None,
+             hideNavBar: Boolean = false
+           )(
+             contentBlock: Html
+           )(implicit
+             request: Request[_],
+             messages: Messages
+           ): HtmlFormat.Appendable
+}
+
+class MainImpl @Inject() (
+                           appConfig: ApplicationConfig,
+                           featureFlagService: FeatureFlagService,
+                           wrapperService: WrapperService,
+                           oldMain: oldMain,
+                           additionalStyles: additionalStylesheets,
+                           additionalScripts: additionalScripts,
+                           ptaHead: PtaHead
+                         ) extends Main with Logging {
+
+  //noinspection ScalaStyle
+  override def apply(
+                      pageTitle: String,
+                      showUrBanner: Boolean,
+                      extendedTitle: Boolean,
+                      optCustomLayout: Option[Html => Html],
+                      sidebar: Option[Html],
+                      hideNavBar: Boolean
+                    )(
+                      contentBlock: Html
+                    )(implicit request: Request[_], messages: Messages): HtmlFormat.Appendable = {
+    val scaWrapperToggle = Await.result(featureFlagService.get(SCAWrapperToggle), Duration(appConfig.scaWrapperFutureTimeout, SECONDS))
+    val fullPageTitle    = if(extendedTitle) {
+      s"$pageTitle - ${Messages("nisp.title.extension")} - GOV.UK"
+    } else pageTitle
+    val trustedHelper    = Try(request.asInstanceOf[AuthenticatedRequest[_]]) match {
+      case Failure(_: java.lang.ClassCastException) => None
+      case Success(value)                           => value.nispAuthedUser.trustedHelper
+      case Failure(exception)                       => throw exception
+    }
+
+    if (scaWrapperToggle.isEnabled) {
+      logger.debug(s"SCA Wrapper layout used for request `${request.uri}``")
+
+      wrapperService.layout(
+        content = contentBlock,
+        pageTitle = Some(fullPageTitle),
+        //serviceNameKey = appConfig.serviceNameKey,
```

Comment:
You are missing this parameter. If you set it in application.conf you don't have welsh.
serviceNameKey = Some(messages("label.service_name"))

The sca template already uses mesasages so you just pass "label.service_name" but I find this very confusing.

Reply:
I don't know how I missed this during my regression or manual testing. The page header is just wrong when in the SCA Wrapper mode. Glad you caught this.